components/notion/sources/new-database/new-database.mjs

```diff
@@ -0,0 +1,45 @@
+import notion from "../../notion.app.mjs";
+import utils from "../common/utils.mjs";
+
+export default {
+  key: "notion-new-database",
+  name: "New Database",
+  description: "Emit new event when a database is created",
+  version: "0.0.1",
+  type: "source",
+  dedupe: "last",
+  props: {
+    db: "$.service.db",
+    timer: {
+      type: "$.interface.timer",
+      default: {
+        intervalSeconds: 60 * 15,
+      },
+    },
+    notion,
+  },
+  async run() {
+    const params = {
+      start_cursor: utils.getLastCursor(this.db),
+    };
+
+    do {
+      const response = await this.notion.listDatabases(params);
```

Comment:
Yes, they are ordered by `created_time`. Still, according to Notion Docs, they may be missed:

> Immediate and complete results. Search indexing is not immediate. If an integration performs a search quickly after a page is shared with the integration (such as immediately after a user performs OAuth), the response may not contain the page. 

In this case, unlike the `Query Database` endpoint, it is not possible to sort or filter by `created_time`.

Another option would be to change dedupe strategy to `unique` and always search for all databases starting from the first cursor. @js07 Please let me know if we should keep it this way or not

Reply:
Since the `unique` [dedupe strategy](https://pipedream.com/docs/components/api/#dedupe-strategies) maintains a list of only the last 100 emitted event `id` values, I think the source would need to maintain its own cache of emitted `id`s (using `$.service.db`) to avoid emitting the same event multiple times.

I think we can keep the source the way it is unless delayed search indexing is likely to caused missed events.